src/main/credentialServer/httpCredentialServer.ts

```diff
@@ -0,0 +1,210 @@
+  }
+
+  async list(helperName: string, data: string, request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
+    return await this.doNamedCommand(requireJSONOutput, helperName, 'list', data, request, response);
+  }
+
+  async erase(helperName: string, data: string, request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
+    return await this.doNamedCommand(requireNoOutput, helperName, 'erase', data, request, response);
+  }
+
+  async store(helperName: string, data: string, request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
+    return await this.doNamedCommand(requireNoOutput, helperName, 'store', data, request, response);
+  }
+
+  protected combineStrings(stdout: string, stderr: string): string {
+    if (!stdout) {
+      return stderr;
+    } else if (!stderr) {
+      return stdout;
+    } else {
+      return `${ stdout }\n\n${ stderr }}`;
+    }
+  }
+
+  protected async doNamedCommand(outputChecker: checkFunctionOutputType, helperName: string, commandName: string, data: string, request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
+    let stdout: string;
+    let stderr: string;
+
+    try {
+      ( { stdout, stderr } = this.runSyncInput(data, helperName, [commandName]));
+      if (outputChecker(stdout, stderr)) {
+        response.writeHead(200, { 'Content-Type': 'text/plain' });
+        response.write(stdout);
+
+        return await Promise.resolve();
+      }
+      stderr = this.combineStrings(stdout, stderr);
+    } catch (err: any) {
+      stderr = this.combineStrings(err.stdout ?? '', err.stderr ?? '');
+    }
+    console.debug(`credentialServer: ${ commandName }: writing back status 400, error: ${ stderr }`);
+    response.writeHead(400, { 'Content-Type': 'text/plain' });
+    response.write(stderr);
+
+    return await Promise.resolve();
+  }
+
+  // Caller is responsible for catching exceptions if this file doesn't exist
+  protected async getCredentialHelperName(): Promise<string> {
+    const home = findHomeDir();
+    const dockerConfig = path.join(home ?? '', '.docker', 'config.json');
+    const contents = JSON.parse((await fs.promises.readFile(dockerConfig, { encoding: 'utf-8' })).toString());
+    const credStore = contents['credsStore'];
+
+    if (!credStore) {
+      throw new Error(`No credStore field in ${ dockerConfig }`);
+    }
+
+    return credStore;
+  }
+
+  closeServer() {
+    this.server.close();
+  }
+
+  protected runSyncInput(data: string, command: string, args: string[]): { stdout: string, stderr: string } {
+    try {
+      // Only the *Sync methods in childProcess that run an external command take a string
+      // directly as input. Setting up a readable stream around a string is a lot more work.
+      const { stdout, stderr } = childProcess.spawnSync(command, args, { input: data, stdio: 'pipe' });
```

Comment:
We should not use `spawnSync` in the main app code (whereas doing it in build scripts is totally fine).

Reply:
Avoided it by using raw`childProcess.spawn`.